### PR TITLE
Improved and added documentation

### LINKED ISSUE #112: Improve documentation and tidy up
## Details:
Hours Spent: 0
Hours estimated: 16

## Acceptance Criteria
- [x] There is a summary of syntax and supported functions available from near the expression input box.
- [x] There is a top-down documentation of the internal design, ie components, modules, state held in components, messages.
- [x] Diagram(s) showing top level decomposition of UI and state.
- [x] State machine showing main user interaction, ie enter expression, set accuracy, press button X, drag...
- [x] How to use the benchmark
- [x] How to profile the application, especially computation of robust plots

## Closing Notes

Overview of changes can be found in PR #135

### PR DESCRIPTION
**Pull request by RelativeForce**
# Issues
works towards #112 

# Summary of changes
- Added graphs images to `README.md`
- Moved the developer instructions to `dev.md`
- Added `benchmark.md` to document the benchmarking process
- Added `syntax.md` to document the expression syntax
- Added link information icon to syntax documentation next to each expression input (Note: these links will only work when the code is merged into master)
- Replaced `Int` usages when `Id` should have been used

# Tests
N/A

# Benchmarks
N/A

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,34 +1,20 @@
...
-## Run tests
-```sh
-npm run test
-```
-This will run all of the unit tests for the project. Note that any functionality that is dependent on external JS libraries will not work in a unit test environment as these libraries are not resolved in the PS test environment.
\ No newline at end of file
+## Further documentation
+- [Expression syntax](/docs/syntax)
+- [Developer instructions](/docs/dev)
+- [Benchmarking](/docs/benchmark)
```
Comment (michalkonecny) [inline README.md]:
```suggestion
- [Benchmarking](/docs/benchmark)
- Design overview diagrams: 
  - [Internal state and component messages](https://drive.google.com/file/d/13J6XdY_bSFnk4_agCbaF5heJJpYJb5nI/view?usp=sharing)
  - [Data flow](https://drive.google.com/file/d/1jMKaTzwTjlZVNtuYDbEFOlKuLuhaBCNn/view?usp=sharing)
  - [UI states](https://drive.google.com/file/d/1TQAOJ9z4zPL4hxErC9RD7v0HPtfWOAvG/view?usp=sharing)
```

File: docs/syntax.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Expression Syntax
+
+In robust plots the expression input only certain input text is accepted. Each expression input is dependent on a single variable. For function plots it is `x` and for the parametric plots it is `x` or `y`. This doc explains the syntax of expressions.
```
Comment (michalkonecny) [inline docs/syntax.md]:
```suggestion
In robust plots the expression input only certain input text is accepted. Each expression input is dependent on a single variable. For function plots it is `x` and for the parametric plots it is `t`. This doc explains the syntax of expressions.
```

File: docs/syntax.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+There are constant values that can be used as operands in the expressions.
+- `pi` : Pi
+- `e` : e
+
+## Unary values
+Unary operators are prefix notation and depending on the function expect there operand to be wrapped in parentheses.
+- `-` : Negation [`-(20) = -20`]
+- `abs` : Absolute [`abs(-25) = 25`]
+- `sqrt` : Square root [`sqrt(25) = 5`]
+- `e^` : Exponent
```
Comment (michalkonecny) [inline docs/syntax.md]:
```suggestion
- `e^` : Natural exponential
```

File: src/Components/ExpressionInput/FunctionExpressionInput/FunctionExpressionInput.purs [hunk 4]
```diff
@@ -87,6 +87,11 @@ render state =
                   , onEnterPressActionEvent UpdateExpression
                   , className "form-control"
                   ]
+              , HH.a
+                  [ className "btn btn-info"
+                  , HP.href "https://github.com/michalkonecny/robust-plots/blob/master/docs/syntax.md"
```
Comment (michalkonecny) [inline src/Components/ExpressionInput/FunctionExpressionInput/FunctionExpressionInput.purs:92]: I think it is best if this link would open a new tab/window by default.  It should not replace the plotter app.  The same also in the the parametric expression input.

**Review by michalkonecny — APPROVED** [review]
Overall, very nice, thank you.  Please consider my suggestions before merging.

File: src/Components/ExpressionInput/FunctionExpressionInput/FunctionExpressionInput.purs (hunk 4 above)
Comment (RelativeForce) [reply src/Components/ExpressionInput/FunctionExpressionInput/FunctionExpressionInput.purs:92]: Resolved in latest commit

**Pull request merged**